File: src/Streamarr.Core/Creators/Creator.cs
using System;
using System.Collections.Generic;
using Streamarr.Core.Datastore;
using Streamarr.Core.Profiles.Qualities;

namespace Streamarr.Core.Creators
{
    public class Creator : ModelBase
    {
        // Display metadata
        public string Title { get; set; } = string.Empty;
        public string CleanTitle { get; set; } = string.Empty;
        public string SortTitle { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string ThumbnailUrl { get; set; } = string.Empty;

        // Organization
        public string Path { get; set; } = string.Empty;
        public string RootFolderPath { get; set; } = string.Empty;
        public int QualityProfileId { get; set; }
        public HashSet<int> Tags { get; set; }

        // State
        public bool Monitored { get; set; }
        public CreatorStatusType Status { get; set; }
        public DateTime Added { get; set; }
        public DateTime? LastInfoSync { get; set; }

        // Navigation
        public LazyLoaded<QualityProfile> QualityProfile { get; set; }

        public Creator()
        {
            Tags = new HashSet<int>();
        }
    }
}

File: frontend/src/typings/Creator.ts
import ModelBase from 'App/ModelBase';

export type CreatorStatusType = 0 | 1 | 2;

interface Creator extends ModelBase {
  title: string;
  titleSlug: string;
  description: string;
  thumbnailUrl: string;
  path: string;
  rootFolderPath: string;
  qualityProfileId: number;
  tags: number[];
  monitored: boolean;
  status: CreatorStatusType;
  added: string;
  lastInfoSync: string | null;
}

export interface CreatorLookupResult {
  name: string;
  description: string;
  thumbnailUrl: string;
  channels: CreatorLookupChannel[];
  existingCreatorId?: number;
}

export interface CreatorLookupChannel {
  platform: string;
  platformId: string;
  platformUrl: string;
  title: string;
  description: string;
  thumbnailUrl: string;
}

export default Creator;
